source/daplink/drag-n-drop/flash_manager.c

```diff
@@ -145,6 +150,18 @@ error_t flash_manager_data(uint32_t addr, const uint8_t *data, uint32_t size)
         // flush if necessary
         if (addr >= current_write_block_addr + current_write_block_size) {
             // Write out current buffer
+#ifdef FLASH_PARTIAL_ERASE
+            // Processing for low-speed devices of chip erasure.
+            if ((current_write_block_addr % intf->erase_sector_size(current_write_block_addr)) == 0) {
+                uint32_t wk_sector = current_write_block_addr / intf->erase_sector_size(current_write_block_addr);
+                status = intf->erase_sector(wk_sector);
+                flash_manager_printf("    intf->erase_sector(sector=0x%x) ret=%i\r\n", wk_sector, status);
+                if (ERROR_SUCCESS != status) {
+                    state = STATE_ERROR;
+                    return status;
+                }
+            }
+#endif
```

Comment:
Please remove the old comment ~193 which reads "//TODO - future improvement - erase sector here" since you are implementing this :).